File: java-spring/hexagonal-feed/hexagonal-feed-application/src/main/java/com/example/hexagonalfeed/common/error/ErrorCode.java
package com.example.hexagonalfeed.common.error;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum ErrorCode {

    INTERNAL_SERVER_ERROR(500, "Internal Server Error"),

    FEED_NOT_FOUND(404, "Feed Not Found");

    private final Integer status;
    private final String message;
}

File: kotlin-spring/feed/src/main/kotlin/com/example/feed/global/error/exception/ErrorCode.kt
package com.example.feed.global.error.exception

import com.example.feed.global.error.response.ErrorResponse

enum class ErrorCode(
        override val status: Int,
        override val message: String
): ErrorResponse {

    EXPIRED_TOKEN(401,"Token Expired" ),
    INVALID_TOKEN(401,"Invalid Token"),
    INVALID_PASSWORD(401, "Invalid Password"),

    USER_MISMATCH(403, "User Mismatch"),

    USER_NOT_FOUND(404, "User Not Found"),
    FEED_NOT_FOUND(404, "Feed Not Found"),

    USER_ALREADY_EXISTS(409, "User Already Exists"),

    INTERNAL_SERVER_ERROR(500, "Internal Server Error")
}